packages/node_modules/@node-red/editor-client/src/js/ui/view.js

```diff
@@ -692,17 +690,21 @@ RED.view = (function() {
                     cancel: function() {
                         quickAddActive = false;
                         resetMouseVars();
+                        updateSelection();
+                        hideDragLines();
+                        redraw();
                     },
-                    add: function(type) {
+                    add: function(type,auto) {
                         quickAddActive = false;
+                        if (auto === true) { mouse_mode = RED.state.QUICK_JOINING; }
                         var result = addNode(type);
                         if (!result) {
                             return;
                         }
                         var nn = result.node;
                         var historyEvent = result.historyEvent;
-                        nn.x = point[0];
-                        nn.y = point[1];
+                        nn.x = Math.ceil((point[0] - 50) / gridSize) * gridSize + 50;
```

Comment:
This is a general enhancement - ensures the new nodes are on the existing grid rather than being placed where the mouse is - so they don't then need to be nudged into position.

Reply:
I think it should only apply this if snap-to-grid is enabled. You can check its value with the rather clunky:
```
if (RED.settings.get("editor").view['view-snap-grid']) {
   ...
}
```

Reply:
can do - but I'd argue that if not using the grid then it'll only ever be gridsize/2 (aka about 10 pixels away from where it would be max... - not that the user knows where the "origin" of the node is anyway (and is behind the select box anyway)...